cylc/flow/rundb.py

```diff
@@ -301,7 +309,6 @@ class CylcSuiteDAO:
             ["name", {"is_primary_key": True}],
             ["flow_label", {"is_primary_key": True}],
             ["status"],
-            ["satisfied"],
```

Comment:
Not sure what are the implications of removing the `satisfied` state from the task pool checkpoints table. Does there need to be a `task_prerequisites_checkpoints` table?

Reply:
> Does there need to be a task_prerequisites_checkpoints table?

In order to preserve checkpointing functionality we would need a checkpoints table subject to the same checkpointing logic as the task pool table.

I have to say I'm not sure I really understand the purpose of checkpointing, especially post Cylc6/SoD so I've put up an issue, partly to bring the number of open issues back up to the golden 256, but mostly to debate the relevance of checkpointing to Cylc8.

My suggestion is, don't add checkpointing in this PR. Await the resolution of that issue, if we need to keep checkpointing we can bring it back in a later PR.

Reply:
(I've commented on checkpoint in the new issue).

Reply:
(We've agreed to removed named checkpoints - well, as good as, no objections so far - so this PR can forget about them).

Reply:
I'll add a comment to the checkpoint issue...